src/coreclr/System.Private.CoreLib/src/Internal/Runtime/Binder/AssemblyName.cs

```diff
@@ -0,0 +1,389 @@
+                &pvPublicKeyToken,  // [OUT] Pointer to the PublicKeyToken blob.
+                &dwPublicKeyToken,  // [OUT] Count of bytes in the PublicKeyToken Blob.
+                &dwHashAlgId,   // [OUT] Hash Algorithm.
+                &pAssemblyName, // [OUT] Name.
+                &amd,           // [OUT] Assembly MetaData.
+                &dwRefOrDefFlags // [OUT] Flags.
+                );
+
+            {
+                string culture = Encoding.UTF8.GetString(amd.szLocale, string.strlen(amd.szLocale));
+                int index = culture.IndexOf(';');
+                if (index != -1)
+                {
+                    culture = culture[..index];
+                }
+
+                CultureOrLanguage = culture;
+                IdentityFlags |= AssemblyIdentityFlags.IDENTITY_FLAG_CULTURE;
+            }
+
+            {
+                string assemblyName = Encoding.UTF8.GetString(pAssemblyName, string.strlen(pAssemblyName));
+
+                const int MAX_PATH_FNAME = 260;
+                if (assemblyName.Length >= MAX_PATH_FNAME)
+                {
+                    throw new Exception("FUSION_E_INVALID_NAME");
+                }
+
+                SimpleName = assemblyName;
+                IdentityFlags |= AssemblyIdentityFlags.IDENTITY_FLAG_SIMPLE_NAME;
+            }
+
+            // See if the assembly[def] is retargetable (ie, for a generic assembly).
+            if ((dwRefOrDefFlags | CorAssemblyFlags.afRetargetable) != 0)
+            {
+                IdentityFlags |= AssemblyIdentityFlags.IDENTITY_FLAG_RETARGETABLE;
+            }
+
+            // Set ContentType
+            if ((dwRefOrDefFlags | CorAssemblyFlags.afContentType_Mask) == CorAssemblyFlags.afContentType_Default)
+            {
+                ContentType = System.Reflection.AssemblyContentType.Default;
+            }
+            else
+            {
+                // We no longer support WindowsRuntime assembly.
+                throw new Exception("FUSION_E_INVALID_NAME");
+            }
+
+            // Set the assembly version
+            {
+                Version = new AssemblyVersion
+                {
+                    Major = amd.usMajorVersion,
+                    Minor = amd.usMinorVersion,
+                    Build = amd.usBuildNumber,
+                    Revision = amd.usRevisionNumber
+                };
+
+                IdentityFlags |= AssemblyIdentityFlags.IDENTITY_FLAG_VERSION;
+            }
+
+            // Set public key and/or public key token (if we have it)
+            if (dwPublicKeyToken != 0 && pvPublicKeyToken != null)
+            {
+                if ((dwRefOrDefFlags & CorAssemblyFlags.afPublicKey) != 0)
+                {
+                    byte* pByteToken;
+                    uint dwTokenLen;
+                    StrongNameTokenFromPublicKey(pvPublicKeyToken, dwPublicKeyToken, &pByteToken, &dwTokenLen);
```

Comment:
We have `AssemblyNameHelpers.ComputePublicKeyToken` for this. No need to call unmanaged code.

Also, we ignore the strong name key for binding purposes, so we may be able to skip eagerly computing the public key token in the first place.